### ISSUE TITLE
AMOS-Style Adaptive Weight Decay as Potential Experimental Feature

### ISSUE THREAD

**Issue by JHawkley** [LoganBooker/prodigy-plus-schedule-free#43]
I've been having a great time with your project here, potentially finding something to replace my work-horse optimizer, [AMOS](https://arxiv.org/abs/2210.11693), for SDXL LoRA training; I'm still running various experiments.  It does make me wonder if AMOS' adaptive weight decay could be a nice fit for this enhanced variant of Prodigy.

Unfortunately, I don't know enough about optimizers to know if would be a fit, so consider this as an, "in case you've not checked this out," kind of thing.  However, if it does fit in, it would be another hyper-parameter that can get tuned automatically, further reducing the burden on the user to find optimal values for their use-case.
